libtock/ipc.c

```diff
@@ -30,9 +30,6 @@ int ipc_register_service_callback(const char *pkg_name,
 }
 
 int ipc_register_client_callback(int svc_id, subscribe_upcall callback, void *ud) {
-  if (svc_id <= 0) {
```

Comment:
Won't this now allow negative values? Why not make `svc_id` a uint

Reply:
Yes, a `uint32_t` is better suited for this purpose. We should change it everywhere though to be consistent.

Reply:
Actually I've changed it to a `size_t`. While the ABI is currently only specified for 32-bit platforms, the kernel does represent the process IDs as `usize`s underneath, so it appears to make sense to use the native integer width in userspace as well.